apps/src/services/services.ts

```diff
@@ -22,107 +22,243 @@ import {
 import {
 	FrequencyType,
 	InteractiveRegionOption,
+	S2DFrequencyType,
 } from '@/types/climate-variable-interface';
-import { LocationS2DData } from '@/lib/s2d';
+import {
+	getApiFrequencyName,
+	getApiVariableId,
+	LocationS2DData,
+} from '@/lib/s2d';
 
 // Cache for API responses to avoid duplicate requests
-const apiCache = new Map<string, any>();
+const apiCache = new Map<string, unknown>();
 
 /**
- * Fetches WordPress variable data by post ID from the custom WP REST API endpoint.
- * Processes and normalizes the response into a structured MapInfoData object,
- * including sectors, trainings, featured image, and dataset taxonomy terms.
+ * Checks if a fetch request was aborted, using its fetchOptions.
+ *
+ * @param fetchOptions - The fetchOptions object passed to the fetch request.
+ *   Can be undefined.
  */
-export const fetchWPData = async (postId: number, fetchOptions?: FetchOptions) => {
-	try {
-		const cacheKey = `wp_data_${postId}`;
+const isAborted = (fetchOptions?: FetchOptions): boolean => {
+	return fetchOptions?.signal?.aborted ?? false;
+}
 
-		// Check cache first
-		if (apiCache.has(cacheKey)) {
-			return apiCache.get(cacheKey);
-		}
+/**
+ * Error returned by fetch functions of this module in case of a request error.
+ */
+export class FetchError extends Error {}
+
+/**
+ * Send a GET JSON request and return the parsed JSON response.
+ *
+ * If the request is aborted (e.g. through a `signal` in `fetchOptions`), the
+ * function will return `null` immediately. No error will be thrown.
+ *
+ * @param url - The URL to fetch. GET parameters can be specified in the
+ *              `params` parameter.
+ * @param params - GET parameters to set in the URL.
+ * @param fetchOptions - Additional options to pass to the fetch request.
+ * @returns A promise that resolves to the parsed JSON response. If the request
+ *     is aborted, the promise resolves to `null`.
+ * @throws FetchError - If the request failed.
+ */
+export const fetchJSON = async <T = any>(
+	url: string,
+	params?: Record<string, string>,
+	fetchOptions?: FetchOptions,
+): Promise<T | null> => {
+	let fetchUrl = url;
+	let response: Response;
 
-		// Make the Fetch request.
-		const response = await fetch(
-			`${WP_API_DOMAIN}${WP_API_VARIABLE_PATH}?post_id=${postId}`,
+	if (params) {
+		fetchUrl += `?${new URLSearchParams(params).toString()}`;
```

Comment:
Ah. I was about to tell you about the benefit of `URLSearchParams`, and it's not removed. Great!